eox_nelp/third_party_auth/pipeline.py

```diff
@@ -143,3 +145,30 @@ def disallow_staff_superuser_users(  # pylint: disable=unused-argument
             )
         )
     return {}
+
+
+def validate_national_id_and_associate_user(request, backend, uid, *args, user=None, social=None, **kwargs):
+    """
+    Validates the user's national ID against the provided SAML UID before associating
+    a SAML identity with a Django user. If validation fails, the session is ended, and
+    the user is redirected to registration.
+
+    Args:
+        request (HttpRequest): The HTTP request object.
+        backend: The authentication backend used, such as SAML.
+        uid (str): Unique identifier from SAML (e.g., user ID).
+        user (User, optional): Django user instance, if found.
+        social (optional): Existing social authentication data, if found.
+
+    Returns:
+        If the UID validation succeeds, proceeds to associate the user with social auth.
+        Otherwise, logs out the current session and redirects to the registration page.
+    """
+    national_id = user.extrainfo.national_id if user and hasattr(user, "extrainfo") else ""
+
+    if national_id and uid.endswith(national_id):
+        return associate_user(backend, uid, user, social, *args, **kwargs)
+
+    logout(request)
+
+    return redirect("/register")
```

Comment:
Here is a redirect. Do you think this pipe would have `@partial `decorator? To continue the process in this step pipe???
https://python-social-auth.readthedocs.io/en/latest/pipeline.html#partial-pipeline

Reply:
That decorator doesn't fit in this context at all,  when I was developing this pipe I included this `backend.strategy.clean_partial_pipeline(partial_token)` to restart the current partial then I realized that is not necessary because `logout(request)` performs that operation, basically this pipe aims to restart the whole pipeline when the national id condition is not valid in order to start a new process so doesn't make sense to start  a new process from this step